### PR TITLE
add new function `nflverse_game_id()`

### PR DESCRIPTION
**Pull request by mrcaseb**
I am undecided about the inout check of vectors with different lengths (it is commented out atm). It is easy to makse sure that all vectors must have the same length. However, we wouldn't allow recycling of single values which is kinda nice.

### CODE REVIEW

**Comment by mrcaseb** [comment]
I expect at least one failing test related to #146

File: R/nflverse_game_id.R [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#' nflverse_game_id(2022, 2, "LAC", "KC")
+nflverse_game_id <- function(season,
+                             week,
+                             away,
+                             home){
+  # catch character input of season and week
+  season <- as.numeric(season)
+  week <- as.numeric(week)
+
+  if (!all(season %in% 1999:most_recent_season())){
```
Comment (tanho63) [inline R/nflverse_game_id.R:20]: most_recent_season(TRUE) perhaps

File: R/nflverse_game_id.R [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  if (!all(away %in% team_abbr_mapping)){
+    cli::cli_abort("{.val {away}} is not a valid {.arg away} abbreviation.")
+  }
+
+  # This could be used to require equal length of input vectors but it
+  # wouldn't allow recycling of single values. So I keep it commented for now
+
+  # if (lengths(list(season, week, away, home)) |> unique() |> length() > 1){
```
Comment (tanho63) [inline R/nflverse_game_id.R:39]:
```r
  arg_lengths <- lengths(list(season,week,away,home))

  if (!all(arg_lengths %in% c(1,max(arg_lengths)))){
    cli::cli_warn("Arguments should be of of length one or of length {max(arg_lengths)}. Recycling may produce unexpected results.")
  }
```

File: R/nflverse_game_id.R [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#' Compute nflverse Game Identifiers
```
Comment (tanho63) [inline R/nflverse_game_id.R:1]: file should be called utils_gameid.R to order with other files

File: tests/testthat/test-nflverse.R [hunk 4]
```diff
@@ -338,3 +338,36 @@ test_that("Cache clearing works",{
...
+    nflverse_game_id(2022, 30, "LAC", "KC"), "must be between 1 and"
+  )
+
+  # error on away team
+  expect_error(
+    nflverse_game_id(2022, 2, "LACC", "KC"), "is not a valid `away` abbreviation"
+  )
+
+  # error on home team
+  expect_error(
```
Comment (tanho63) [inline tests/testthat/test-nflverse.R:369]: add test for funny lengths

File: R/nflverse_game_id.R (hunk 1 above)
Comment (mrcaseb) [reply R/nflverse_game_id.R]: Done

File: R/nflverse_game_id.R (hunk 2 above)
Comment (mrcaseb) [reply R/nflverse_game_id.R]: done, thanks!

File: tests/testthat/test-nflverse.R (hunk 4 above)
Comment (mrcaseb) [reply tests/testthat/test-nflverse.R:369]: done

File: R/nflverse_game_id.R (hunk 3 above)
Comment (mrcaseb) [reply R/nflverse_game_id.R]: done

**Review by tanho63 — APPROVED** [review]
Lgtm

**Pull request merged**